ch13/ex13_08.h

```diff
@@ -18,11 +18,13 @@ class HasPtr {
 public:
     HasPtr(const std::string &s = std::string()) : ps(new std::string(s)), i(0) { }
     HasPtr(const HasPtr &hp) : ps(new std::string(*hp.ps)), i(hp.i) { }
-    HasPtr& operator=(const HasPtr &hp) {
-        std::string *new_ps = new std::string(*hp.ps);
-        delete ps;
-        ps = new_ps;
-        i = hp.i;
+    HasPtr& operator=(const HasPtr &rhs_hp) {
+        if(this != &rhs_hp){
+            std :: string *temp_ps = new std :: string(*rhs_hp);
```

Comment:
In addition, `new std :: string(*rhs_hp);` cannot pass compiler, cause `*rhs_hp` is not a valid parameter of std::string.